scripts/test-e2e-local.js

```diff
@@ -52,202 +42,213 @@ const argv = yargs
     alias: 'hermes',
     type: 'boolean',
     default: true,
+  })
+  .option('c', {
+    alias: 'circleciToken',
+    type: 'string',
+    required: true,
```

Comment:
I don't think it should be required. If not provided, it should fall back to generating the artifacts locally. I can see scenarios in which we will need to be able to generate the artifacts locally.

Reply:
Can you elaborate with those scenarios? We had a similar discussion internally and we ended up considering that there are no meaningful scenarios for which we want to regenerate stuff locally as that will not be representative of the actual release.

If we make it required, we can remove the fallback and simplify the script by a lot.

Reply:
> then once the commit has completed all of its CI, and only then, run the local test script. I don't think we should prevent being able to test stuff right away 

It won't make much of a difference, actually. Building `RNTestProject` locally takes more or less the same time! 😓 So I'd rather wait for CI and have my machine free from building load to do something else while CI builds.

> what does happen if you are trying to run the script before the CI finishes?

Right now, it emits an error says that the job has to finish.

> 2. I want to try and build the apk with a different setup from the one used on CI

This makes sense... ok, I have meetings now... but tomorrow I'll restore the old path also.

But the script is becoming complex so, for my sanity, I'm going to split it in functions for better readability.